docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,207 @@
+
+.. code-block:: python
+
+  from multiprocessing import cpu_count
+  import modin.config as cfg
+  import modin.pandas as pd
+  import numpy as np
+  import timeit
+
+  # Generating data for a square-like frame
+  data = np.random.randint(0, 100, size=(2**10, 2**10))
+
+  # Explicitly setting `NPartitions` to its default value
+  cfg.NPartitions.put(cpu_count())
+
+  # Each worker process `sqrt(cpu_count())` amount of partitions
+  df = pd.DataFrame(data)
+  print(f"10 times of .abs(): {timeit.timeit(lambda: df.abs(), number=10)}s.")
+  # Possible output: 2.59s.
+
+  # Taking a square root of the the current `NPartitions` to make more even partitioning
+  cfg.NPartitions.put(int(cpu_count() ** 0.5))
+
+  # Each worker process a single partition
+  df = pd.DataFrame(data)
+  print(f"10 times of .abs(): {timeit.timeit(lambda: df.abs(), number=10)}s.")
+  # Possible output: 0.24s.
+
+Do not iterate over Modin DataFrame
+"""""""""""""""""""""""""""""""""""
+
+Use ``df.apply()`` or other aggregation methods when possible instead of iterating over a frame.
+For-loops don't scale and force to collect all the distributed data to the driver.
+
+Copy-pastable example, showing how replacing a for-loop to the equivalent ``.apply()`` may improve performance:
+
+.. code-block:: python
+
+  import modin.pandas as pd
+  import numpy as np
+  from timeit import default_timer as timer
+
+  data = np.random.randint(1, 100, (2 ** 10, 2 ** 2))
+
+  md_df = pd.DataFrame(data)
+
+  result = []
+  t1 = timer()
+  # Iterating over a frame forces to collect distributed data to the driver and doesn't scale
+  for idx, row in md_df.iterrows():
+      result.append((row[1] + row[2]) / row[3])
+  print(f"Filling a list by iterating a Modin frame: {timer() - t1:.2f}s.")
+  # Possible output: 36.15s.
+
+  t1 = timer()
+  # Using `.apply()` perfectly scales to all axis-partitions
+  result = md_df.apply(lambda row: (row[1] + row[2]) / row[3], axis=1).to_numpy().tolist()
+  print(f"Filling a list by using '.apply()' and converting the result to a list: {timer() - t1:.2f}s.")
+  # Possible output: 0.22s.
+
+Use Modin's algebra API to implement custom parallel functions
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Modin provides a set of low-level parallel-implemented operators which can be used to build most of the
+aggregation functions. These operators are present in the :doc:`algebra module </flow/modin/core/dataframe/algebra>`.
+Modin DataFrame allows for users to use their own aggregations built with this module. Visit the 
+:doc:`appropriate section </flow/modin/core/dataframe/algebra>` of the documentation for the steps to do it.
+
+Do not mix pandas and Modin DataFrames
+""""""""""""""""""""""""""""""""""""""
+
+Although Modin is considered to be a drop-in replacement for pandas, they are not intended to be used together
+in a single flow. Passing a pandas DataFrame as an argument for a Modin's frame method may either slowdown
```

Comment:
Is there a reason we're repeatedly referring to a dataframe as a frame? (I changed a few instances earlier.)

Reply:
replaced "frame" -> "dataframe"